File: library_arouter/src/main/java/cn/jack/library_arouter/manager/ArouterManager.java
package cn.jack.library_arouter.manager;

import com.alibaba.android.arouter.launcher.ARouter;

import cn.jack.library_arouter.router.RouterPathActivity;
import cn.jack.library_arouter.router.RouterPathFragment;

/**
 * @创建者 Jack
 * @创建时间 2021/3/4 17:07
 * @描述
 */
public class ArouterManager {

    /**
     * 使用静态内部类的方式实现单例
     */
    public static ArouterManager getInstance() {
        return ArouterManager.Holder.INSTANCE;
    }

    private static class Holder {
        private static final ArouterManager INSTANCE = new ArouterManager();
    }

    private ArouterManager() {

    }

    public void navigation2Home(){
        ARouter.getInstance().build(RouterPathActivity.Home.PAGER_HOME).navigation();
    }

    public void navigation2Login(){
        ARouter.getInstance().build(RouterPathActivity.Login.PAGER_LOGIN).navigation();
    }

    public void navigation2Register(){
        ARouter.getInstance().build(RouterPathActivity.Register.PAGER_REGISTER).navigation();
    }

    public void navigation2HomeFragment01(){
        ARouter.getInstance().build(RouterPathFragment.HomeFirst.PAGER_HOME_FIRST).navigation();
    }
}

File: library_arouter/src/main/java/cn/jack/library_arouter/manager/ArouterManager.kt
package cn.jack.library_arouter.manager

import android.os.Bundle
import com.alibaba.android.arouter.launcher.ARouter

/**
 * @创建者 Jack
 * @创建时间 2021/3/4 17:07
 * @描述
 */
class ArouterManager private constructor() {
    companion object {
        @Volatile
        private var instance: ArouterManager? = null

        fun getInstance(): ArouterManager {
            if (instance == null) {
                synchronized(this) {
                    if (instance == null) {
                        instance = ArouterManager()
                    }
                }
            }

            return instance!!
        }
    }

    fun navigationTo(toPath: String, bundle: Bundle? = null) {
        if (bundle != null) {
            ARouter.getInstance().build(toPath)
                .with(bundle)
                .navigation()
        } else {
            ARouter.getInstance().build(toPath)
                .navigation()
        }
    }
}